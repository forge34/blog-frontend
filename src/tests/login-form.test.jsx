import { render, screen } from "@testing-library/react";
import LoginForm from "../components/login-form";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";

describe("Login form test", () => {
  it("should match snapshot", () => {
    const { form } = render(
      <MemoryRouter>
        <LoginForm></LoginForm>
      </MemoryRouter>,
    );

    expect(form).toMatchSnapshot();
  });

  it("should render submit button", () => {
    render(
      <MemoryRouter>
        <LoginForm></LoginForm>
      </MemoryRouter>,
    );

    expect(screen.getByRole("button")).toHaveTextContent("Login");
  });

  it("should navigate on submit", async () => {
    render(
      <MemoryRouter>
        <LoginForm></LoginForm>
      </MemoryRouter>,
    );

    const user = userEvent.setup();

    await user.click(screen.getByRole("button"));

    expect(location.pathname).toMatch("/");
  });
});
