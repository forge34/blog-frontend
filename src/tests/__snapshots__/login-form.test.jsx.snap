// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login form test > should match snapshot 1`] = `undefined`;
